### PR TITLE
massive testing & repo overhaul

### PR DESCRIPTION
**Pull request by PatrickAlphaC**
- Minimized & cleaned tests
- Created event listeners instead of relying on `sleep` for running tests
- Added gas reporter
- Cleaned up readme
- Condensed task imports
- Added auto-verification
- Removed `99_Setup_Contracts` in favor of adding auto fund directly with `AUTO_FUND` environment variable
- Renamed `integration` to `staging`
- Added events to `APIConsumer` and `RandomNumberConsumer`
- Added `solhint` script
- Removed `eslint` in favor of just a `prettier` formatter
- Added a sample script over tasks

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -1,125 +1,124 @@
...
+  "https://eth-mainnet.alchemyapi.io/v2/your-api-key"
+const RINKEBY_RPC_URL =
+  process.env.RINKEBY_RPC_URL || "https://eth-rinkeby.alchemyapi.io/v2/your-api-key"
 const KOVAN_RPC_URL = process.env.KOVAN_RPC_URL || "https://eth-kovan.alchemyapi.io/v2/your-api-key"
-const POLYGON_MAINNET_RPC_URL = process.env.POLYGON_MAINNET_RPC_URL || "https://polygon-mainnet.alchemyapi.io/v2/your-api-key"
-const MNEMONIC = process.env.MNEMONIC || "your mnemonic"
-// optional
+const POLYGON_MAINNET_RPC_URL =
+  process.env.POLYGON_MAINNET_RPC_URL || "https://polygon-mainnet.alchemyapi.io/v2/your-api-key"
 const PRIVATE_KEY = process.env.PRIVATE_KEY || "your private key"
```
Comment (andrejrakic) [inline hardhat.config.js:24]: If we want to close issue #74 we should 
```suggestion
const PRIVATE_KEY = process.env.PRIVATE_KEY
```

and then in each `networks` object make private key optional like this
```
accounts: PRIVATE_KEY !== undefined ? [PRIVATE_KEY] : []
```

File: hardhat.config.js [hunk 2]
```diff
@@ -1,125 +1,124 @@
...
-		},
+    localhost: {
+      chainId: 31337,
     },
-    gasReporter: {
-        enabled: process.env.REPORT_GAS ? true : false,
-        currency: "USD",
+    kovan: {
+      url: KOVAN_RPC_URL,
+      accounts: [PRIVATE_KEY],
```
Comment (andrejrakic) [inline hardhat.config.js:48]:
```suggestion
      accounts: PRIVATE_KEY !== undefined ? [PRIVATE_KEY] : [],
```

File: hardhat.config.js [hunk 3]
```diff
@@ -1,125 +1,124 @@
...
+      // },
+      saveDeployments: true,
+      chainId: 42,
     },
-    contractSizer: {
-        runOnCompile: false,
-        only: ["APIConsumer", "KeepersCounter", "PriceConsumerV3", "RandomNumberConsumer"],
+    rinkeby: {
+      url: RINKEBY_RPC_URL,
+      accounts: [PRIVATE_KEY],
```
Comment (andrejrakic) [inline hardhat.config.js:57]:
```suggestion
      accounts: PRIVATE_KEY !== undefined ? [PRIVATE_KEY] : [],
```

File: hardhat.config.js [hunk 4]
```diff
@@ -1,125 +1,124 @@
...
-        deployer: {
-            default: 0, // here this will by default take the first account as deployer
-            1: 0 // similarly on mainnet it will take the first account as deployer. Note though that depending on how hardhat network are configured, the account 0 on one network can be different than on another
-        },
-        feeCollector: {
-            default: 1
-        }
+    mainnet: {
+      url: MAINNET_RPC_URL,
+      accounts: [PRIVATE_KEY],
```
Comment (andrejrakic) [inline hardhat.config.js:66]:
```suggestion
      accounts: PRIVATE_KEY !== undefined ? [PRIVATE_KEY] : [],
```

File: hardhat.config.js [hunk 5]
```diff
@@ -1,125 +1,124 @@
...
-            {
-                version: "0.6.6"
-            },
-            {
-                version: "0.4.24"
-            }
-        ]
+    polygon: {
+      url: POLYGON_MAINNET_RPC_URL,
+      accounts: [PRIVATE_KEY],
```
Comment (andrejrakic) [inline hardhat.config.js:75]:
```suggestion
      accounts: PRIVATE_KEY !== undefined ? [PRIVATE_KEY] : [],
```

File: tasks/withdraw-link.js [hunk 6]
```diff
@@ -1,36 +1,43 @@
-const { networkConfig, getNetworkIdFromName } = require('../helper-hardhat-config')
+const { networkConfig } = require("../helper-hardhat-config")
```
Comment (andrejrakic) [inline tasks/withdraw-link.js:1]: Should we delete this file, or is this leftover?

File: test/unit/APIConsumer_unit_test.js [hunk 7]
```diff
@@ -1,63 +1,65 @@
...
-      }
+      })
 
-      const MockOracle = await deployments.get("MockOracle");
-      mockOracle = await ethers.getContractAt("MockOracle", MockOracle.address);
-    })
+      it("Should successfully make an API request", async () => {
+        const transaction = await apiConsumer.requestVolumeData()
+        const transactionReceipt = await transaction.wait(1)
+        const requestId = transactionReceipt.events[0].topics[1]
```
Comment (andrejrakic) [inline test/unit/APIConsumer_unit_test.js:26]:
```solidity
const requestId = await apiConsumer.callStatic.requestVolumeData()
```
is cleaner imo, we should add this in the next PR

**Review by andrejrakic — APPROVED** [review]
LGTM! This PR improved our quality of js development life 🚀

File: test/unit/APIConsumer_unit_test.js (hunk 7 above)
Comment (PatrickAlphaC) [reply test/unit/APIConsumer_unit_test.js:26]: Agreed.

**Pull request merged**